docker.go

```diff
@@ -83,6 +84,35 @@ func (p Plugin) Exec() error {
 		time.Sleep(time.Second * 1)
 	}
 
+	// Create Auth Config File
+	if p.Login.AuthConfig != "" {
+		fmt.Println("AuthConfig provided.")
+		err_mkdir := os.MkdirAll("/root/.docker", 0600)
+		if err_mkdir != nil {
+			fmt.Println("Error creating root's docker auth config directory: ")
+			fmt.Println(err_mkdir)
+		}
+		cfile, err_create := os.Create("/root/.docker/config.json")
+		if err_create != nil {
+			fmt.Println("Error creating root's docker auth config file: ")
+			fmt.Println(err_create)
+			cfile.Close()
+		}
+		err_chmod := os.Chmod("/root/.docker/config.json", 0600)
+		if err_chmod != nil {
+			fmt.Println("Error setting permissions on root's docker auth config file: ")
+			fmt.Println(err_chmod)
+		}
+		_, err_str := cfile.WriteString(p.Login.AuthConfig)
+		if err_str != nil {
+			fmt.Println("Error filling root's docker auth config file: ")
+			fmt.Println(err_str)
+		} else {
+			cfile.Close()
+		}
```

Comment:
we can combine these three commands into a single command:

```
ioutil.WriteFile("/root/.docker/config.json", []byte(p.Login.AuthConfig, 0600)
```

Reply:
Perfect I'll test it immediatly